client/service/src/builder.rs

```diff
@@ -128,7 +138,7 @@ pub struct ServiceBuilder<TBl, TRtApi, TGen, TCSExt, TCl, TFchr, TSc, TImpQu, TF
 	remote_backend: Option<Arc<dyn RemoteBlockchain<TBl>>>,
 	marker: PhantomData<(TBl, TRtApi)>,
 	background_tasks: Vec<(&'static str, BackgroundTask)>,
-	prometheus_registry: Option<Registry>
+	prometheus_registry_and_port: Option<(Registry, SocketAddr)>,
```

Comment:
I'd suggest adding separate data structure for this, `PrometheusLink` or similar

Reply:
which already exists, `PrometheusConfig` ?

Reply:
Done.